core/src/main/java/com/amplitude/core/utilities/EventsFileManager.kt

```diff
@@ -181,23 +251,92 @@ class EventsFileManager(
     }
 
     // write to underlying file
-    private fun writeToFile(content: ByteArray, file: File) {
-        FileOutputStream(file, true).use {
-            it.write(content)
-            it.flush()
+    private fun writeToFile(
+        content: ByteArray,
+        file: File,
+        append: Boolean = true,
+    ) {
+        try {
+            FileOutputStream(file, append).use {
+                it.write(content)
+                it.flush()
+            }
+        } catch (e: FileNotFoundException) {
+            diagnostics.addErrorLog(("Error writing to file: ${e.message}"))
+            logger.error("File not found: ${file.path}")
+        } catch (e: IOException) {
+            diagnostics.addErrorLog(("Error writing to file: ${e.message}"))
+            logger.error("Failed to write to file: ${file.path}")
+        } catch (e: SecurityException) {
+            diagnostics.addErrorLog(("Error writing to file: ${e.message}"))
+            logger.error("Security exception when saving event: ${e.message}")
+        } catch (e: Exception) {
+            diagnostics.addErrorLog(("Error writing to file: ${e.message}"))
+            logger.error("Failed to write to file: ${file.path}")
         }
     }
 
-    private fun writeToFile(content: String, file: File) {
-        file.createNewFile()
-        FileOutputStream(file).use {
-            it.write(content.toByteArray())
-            it.flush()
+    private fun writeEventsToSplitFile(
+        events: List<JSONObject>,
+        file: File,
+        append: Boolean = true,
+    ) {
+        val contents = events.joinToString(separator = "\n", postfix = "\n") { it.toString() }
```

Comment:
Is there any chance of `\n` in the event name or properties?

Reply:
Great question! I believe that's possible at least for event or property values. In this write piece this should be fine, though I am a bit more worried on the read part. Will test that a bit more and see how that goes. BTW, any suggestions for handling that?

Reply:
@justin-fiedler I tried to use the url encode/decode to handle the line break issue and hopefully that helps, let me know your thoughts.

Reply:
Thanks. I think the encoding should fix it but am a little concerned about the extra processing for all events. I wonder if there is some other character we can use for the delimiter instead of `\n` that user's are not capable of entering e.g. some special unicode value?

Reply:
@justin-fiedler I am not sure if there is other values we could use. I guess if we use other value, we still need to handle that value is possibly part of the event or property value. We could discuss later in sync about this.

Reply:
Just updated to use the new delimiter as discussed.